### PR TITLE
Upgrade the following utils:  binv  cmpbqm  gettab  readbb  readmp  sinv

### LINKED ISSUE #91: test codes for utils programs
As noted in the discussion for #88 , all sample programs under /utils need to have an associated test code under /test.  Currently, only gettab has an associated test code, so we still need them for readbp, readmp, binv, sinv, and cmpbqm at some point.

There's no specific deadline for this task; rather, this is just a placeholder for tracking this issue and for any further discussion or developments.

### LINKED ISSUE #141: binv utility does not work as documented
As currently documented, the binv utility is to be run as `binv.x filename`, where `filename` is the name of the file to be inventoried.  However, this doesn't work as advertised, and in this case the executable just hangs in a perpetual waiting state.  This behavior was noted on both hera and on WCOSS/Dell.  Looking at the binv source code, it's doing a `read(5,'(a)') file` to get the filename, and apparently this doesn't work when the filename is a command-line argument.

However, it does work if you just type in the name of the executable `binv.x` and hit \<Enter\>, and then, when it's sitting there waiting for input, paste in the `filename` and hit \<Enter\> again.  If this is the way the utility is really designed to work(?), then we need to update the documentation accordingly in the Sample Utilities section of the documentation, and in this case it might also be helpful for the user to have the executable spit out a prompt such as `Enter [path/]name of file to be inventoried:` to let them know it's waiting for a filename.  Otherwise, and if the intent is really for the `filename` to be passed as a command-line argument, then the code needs to be tweaked a bit.

This is related to #91, given that we currently don't have a test case for this or many of the other Sample Utilities.  @jack-woollen, could you please take a look at this when you have some time?

### PR DESCRIPTION
**Pull request by jack-woollen**
…inv  split_by_subset

1) Ensure they each read inputs as arguments to the executable
2) Ensure each source filename uses the f90 extension
3) Provide a run test for each to pass during a build
4) Remove the .x extension from the executables
5) Update the readme file documentation

### CODE REVIEW

**Comment by jack-woollen** [comment]
Sorry, it probably needs the new tarfile: dogwood:/lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork/bufr-11.6.0.tar

**Comment by jbathegit** [comment]
Thanks Jack, and I'm adding the following markers:

Fixes #91, #141

to this thread, which in turn will automatically close those issues whenever these updates are merged.  But before then, I have some questions:

1. If you look at the merge checks below, you'll see that there were build issues.  If you click on "Details" in the cases with the failing checks, then scroll down to the "Build" section, you'll see there were syntax errors flagged with many of the data statements in sinv.f90.  Those will need to get cleaned up before we can merge this.
2. And speaking of sinv.f90, I'm curious why you need all of these data statements?  Instead of hardcoding all of these code figures and corresponding string values, why not just use the NCEPLIBS-bufr subroutine [getcfmng()](https://noaa-emc.github.io/NCEPLIBS-bufr/getcfmng_8f.html) to get this information?  All you'd need to do is first call subroutine [codflg()](https://noaa-emc.github.io/NCEPLIBS-bufr/codflg_8f.html) to enable this capability, and then you can make any number of successive calls to subroutine getcfmng to access any number of code figures and associated meanings from any of the official WMO code or flag tables.

**Comment by jbathegit** [comment]
Fixes #91

Fixes #141

**Comment by jack-woollen** [comment]
Thanks Jeff. I didn't see those errors on wcoss. Looks like whatever fortran version is used in github has a line length limit. I guess i could use continuations, but what a mess. I think point 2 is neater and cleaner, I'll try that.

**Comment by jack-woollen** [comment]
I tried using getcfmng but it needs to know where the master bufr tables are, which makes sinv much less portable. Simply truncating the data statements to 100 cols should fix the compile problem. That's now pushed to the fork.

**Comment by jack-woollen** [comment]
the codes all compile now but it needs the new tarfile to run the tests

**Comment by jbathegit** [comment]
> I tried using getcfmng but it needs to know where the master BUFR tables are, which makes sinv much less portable. Simply truncating the data statements to 100 cols should fix the compile problem. That's now pushed to the fork.

Yes, you're right it does need to know where the master tables are, and sorry I forgot to mention that.  But it's not difficult to do and still leads to a portable solution; in fact, we already do this with the debufr utility.

Please see the [pending source changes](https://github.com/NOAA-EMC/NCEPLIBS-bufr/pull/185/files) for #185, which we just haven't been able to merge into the `develop` branch yet because issue #186 is still waiting on a fix.  At any rate, the key is to use the global MASTER_TABLE_DIR macro, which is set within CMake and whose value can be embedded in the source code of a utility program such as debufr (which is C code) or sinv (which is Fortran code).

So in this case you would use the macro @MASTER_TABLE_DIR_F@ for sinv (since it's Fortran code), via the following steps:
1. modify the sinv source code to add a new character string variable set to this macro
2. add a call to [mtinfo()](https://noaa-emc.github.io/NCEPLIBS-bufr/mtinfo_8f.html) which includes that new character string variable as argument CMTDIR
3. rename the file sinv.f as sinv.f.in and use the CMake configure_file command to replace this macro with the actual location of the master BUFR tables directory at compile time (see utils/debufr.f.in and utils/CMakeLists.txt for an example of how this is done for the debufr utility)

Then it works and is completely portable, and I would argue is much more preferable to hard-coding all of the SAID and SIID code table values and meaning strings into the sinv source code, which you then have to remember to update whenever WMO adds new entries to those code tables.

I realize this may be a bit confusing if you've never done it before or don't have a lot of experience with CMake, so if you want I'd be happy to go ahead myself and add a new commit to this branch which does this - just let me know ;-)

**Comment by jbathegit** [comment]
I just copied your new tarfile up to the server, so you should be able to test with it now.

But please also consider my offer to help you incorporate the master BUFR tables into your sinv source code, rather than having to keep hardcoded copies of all of the SAID and SIID code table values and meanings as separate data statements.   I really think the former is the best long-term solution, so that sinv will always remain in sync with WMO's updates to those tables.

**Comment by edwardhartnett** [comment]
This looks like far too many changes for one issue and one PR.

Can you break these changes up to separate tickets?

**Comment by jack-woollen** [comment]
@edwardhartnett  Why? All the changes have been debugged already. First in the working copy, and then in the github environment. In the last update all the tests ran but they couldn't validate because the  updated (non-git) tar file wasn't installed yet. so the test output data had nothing to compare with. It is presumably now in place. I understand the point , and I'm happy to do one change at a time in future, but it would be a lot of work now to build these changes back into a clean working copy one at time and wait for each pr to be installed separately in order to do the next one.

**Comment by jack-woollen** [comment]
@jbathegit  So I couldn't find any macro named @MASTER_TABLE_DIR_F@. But the macro name in debufr.c called @MASTER_TABLE_DIR@ is defined in NCEPLIBS-bufr/cmake/PackageConfig.cmake.in as 
set(@PROJECT_NAME@_TABLES_DIR "@MASTER_TABLE_DIR@")

In my working set this defaults to /lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork/build/tables. But what does it default to in the github cmake environment? 

# Test debufr.x utility
set(db_flags_1 "-t ../tables")
set(db_flags_2 "-t testfiles/data -f bufrtab.031 -c")
foreach(db_num RANGE 1 2)
  set(db_case debufr_${db_num})
  add_test(NAME test_${db_case} COMMAND ${CMAKE_BINARY_DIR}/bin/test_debufr.sh
           "${CMAKE_BINARY_DIR}/utils/debufr.x ${db_flags_${db_num}}" "testfiles/data/${db_case}"
           "testrun/${db_case}.run" "testfiles/testoutput/${db_case}.out")
endforeach()

I notice when the debufr test is run (above) the master table locations are hard wired to specific locations, and the macro resolution isn't used for locating tables. Do I need to do something like that to locate the master tables for the sinv test in the github repo?

**Comment by jbathegit** [comment]
@jack-woollen, sorry that this is confusing, but most of that stems from the fact that we still have a bunch of PRs lined up and waiting which can't be merged because of #186.  As noted above, the pending change which contains the new @MASTER_TABLE_DIR_F@ macro is one of these, so for now at least you need to look [here](https://github.com/NOAA-EMC/NCEPLIBS-bufr/pull/185/files) to see that macro.

You won't need to worry about NCEPLIBS-bufr/cmake/PackageConfig.cmake.in; rather, the change we'll need to make to do the macro substitution is in NCEPLIBS-bufr/utils/CMakeLists.txt.  Specifically, we'll just need an additional "configure_file" command, similar to the existing one for debufr.c.in.

As for the "test debufr.x" lines you quoted from NCEPLIBS-bufr/test/CMakeLists.txt, you're correct that you/we'll probably need to do something similar here for the sinv utility, in order to pass in the location of the master tables during the test step.  The reason is because the test step (ctest) takes place **before** the install step (make install), and the latter is when the tables actually get installed into the location defined in the macro.  So, bottom line, anything that tries to read from that location prior to the install step won't find that directory.  But the good news is that, once the install step is done, any future invocation of the sinv utility won't need to repeat the process of passing in the location of the tables, because at that point the default location will be hardcoded into the utility via the macro, so one would only need to pass in a location via the command line if they wanted to override that default location.

Again, I think we can work through all of these issues, and I'm happy to help you get it all sorted it, but I think the first priority is to get #186 squared away, which in turn will allow us to merge #185, and then at that point we'll have the correct @MASTER_TABLE_DIR_F@ macro in place that we can build upon.

**Comment by jack-woollen** [comment]
@jbathegit Since the @MASTER_TABLE_DIR@ macro work just file what's the point of @MASTER_TABLE_DIR_F@?

**Comment by jbathegit** [comment]
> @jbathegit Since the @MASTER_TABLE_DIR@ macro work just file what's the point of @MASTER_TABLE_DIR_F@?

As noted in #185, if @MASTER_TABLE_DIR@ resolved to a directory pathname that was longer than 132 characters, then inserting that string directly into Fortran source code would violate the Fortran standard because the entire pathname would end up on one source code line.

So I came up with a patch to CMakeLists.txt which splits up @MASTER_TABLE_DIR@ into substrings of less than 132 characters, along with the requisite '& ' continuation and linefeed characters needed to spread it among multiple lines of a Fortran source code file, all the while staying under the limit of 132 characters per line.  The result is the new @MASTER_TABLE_DIR_F@ macro, and I went ahead and created a similar one for C as well, but the original @MASTER_TABLE_DIR@ macro is still needed as well for the install step, as well as the basename from which to generate the latter two macros.

**Comment by edwardhartnett** [comment]
PRs that do not follow the quality process will not be merged. There will be no exceptions.

Where are the GitHub issues to describe these changes?

**Comment by jbathegit** [comment]
@edwardhartnett, the relevant issues are linked above - they are #91 and #141, and there's an overlap because the utility mentioned in the second issue is one of the same programs referenced in the first issue.

**Comment by edwardhartnett** [comment]
@jbathegit I will leave it for you to decide how to proceed with this PR.

**Comment by jack-woollen** [comment]
@jbathegit Okay, got it. I'll wait until the repo is caught up before pushing the final (?) changes for these utilities.

**Comment by jack-woollen** [comment]
And thanks for all the comments, very helpful.

File: utils/sinv.f.in [hunk 1]
```diff
@@ -0,0 +1,106 @@
+!-----------------------------------------------------------------------
```
Comment (jbathegit) [inline utils/sinv.f.in]: filename should be sinv.f90.in, not sinv.f.in

File: utils/split_by_subset.f90 [hunk 2]
```diff
@@ -25,15 +25,12 @@ program split_by_subset
...
-  if (trim(adjustl(finput)) == '') finput = 'fort.20'
-
-  !> check if file exists, then open it, else abort
   inquire(file=trim(adjustl(finput)), exist=file_exists)
   if (file_exists) then
     open(lunit, file=trim(adjustl(finput)), form='unformatted')
     call openbf(lunit,'IN',lunit)
   else
-    call bort('File ' // trim(adjustl(finput)) // ' does not exist')
+    call bort('Usage: "split_by_subset bufrfile" will split a BUFR file into subsets')
```
Comment (jbathegit) [inline utils/split_by_subset.f90]: Maybe I'm missing something here, but it seems like the old error message was more appropriate.  The new error message doesn't seem appropriate for when the boolean "file_exists" tests false.

File: utils/sinv.f.in [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+!  call routines to look  up the said and siid 
+
+      if(icode>0) call getcfmng(lunit,'SAID',icode,'  ',-1,csad,len,iret)
+      if(jcode>0) call getcfmng(lunit,'SIID',jcode,'  ',-1,csid,len,iret)
+
+      return
+      end
+!-----------------------------------------------------------------------
+!-----------------------------------------------------------------------
+      SUBROUTINE BORT(STR)
```
Comment (jbathegit) [inline utils/sinv.f.in]: maybe just add a comment here that this version of bort is superseding the library version of the subroutine of the same name

File: utils/binv.f90 [hunk 4]
```diff
@@ -43,18 +49,18 @@ PROGRAM BINV
...
  
-c     IOFF = 1
-c     CALL STATUS(LUNBF,LUN,IL,IM)
-c     call ufbcnt(lunbf,irec,isub)
-c     DO I=1,NMSUB(LUNBF)
-c     NBYS = IUPB(MBAY(1,LUN),MBYT(LUN)+IOFF,16)
-c     print*,SUBSET,' m#',irec,' subt#',i,nbys
-c     IOFF = IOFF+NBYS
-c     ENDDO
+!     IOFF = 1
```
Comment (jbathegit) [inline utils/binv.f90]: Is there any particular reason why we're keeping these 8 commented-out lines of code?

File: utils/CMakeLists.txt [hunk 5]
```diff
@@ -1,21 +1,40 @@
...
+  split_by_subset
+)
+
 foreach(_src ${_utils_srcs})
   get_filename_component(_name "${_src}" NAME_WE)
   get_filename_component(_ext "${_src}" LAST_EXT)
+
   set(_exec "${_name}.x")
+
+  foreach(_nox ${_no_x})
```
Comment (jbathegit) [inline utils/CMakeLists.txt:32]: I like the idea to remove the .x from the executable names.  At some point I'll plan to do the same with the other utilities (debufr, xbfmg, etc.), and then we can just directly set _exec to ${_name} for **all** of the utilities within this file, without continuing to need a _no_x list.  But this is good for now, and again a nice idea!

**Comment by jack-woollen** [comment]
Perhaps. Either way it's saying you need to supply an actual file. What if
for example it reads a blank argument. I could beef it up.
On another related subject, I noticed that ufbtab makes an error processing
a file with msgunp=1, the foreign file case. I made a fix for it.
Should I include that with these changes or do a separate fork?

File: utils/binv.f90 (hunk 4 above)
Comment (jack-woollen) [reply utils/binv.f90]: None. It can be cleaned up

File: utils/sinv.f.in (hunk 1 above)
Comment (jack-woollen) [reply utils/sinv.f.in]: Okay, just following guidance from above.

File: utils/sinv.f.in (hunk 3 above)
Comment (jack-woollen) [reply utils/sinv.f.in]: Its for debugging, I could take it out.

**Comment by jbathegit** [comment]
Thanks again for all of your efforts here @jack-woollen.  Overall everything looks good, and I just had a few comments/questions as noted above in the thread, and which I see you've already begun replying to.

A bigger issue is that GitHub doesn't seem to be able to automatically merge this new branch with the updated develop branch, presumably because something in one of the other PRs directly conflicts with a change in this new branch.  So those conflicts may need to be resolved outside of GitHub, by manually merging or rebasing this new branch with the updated develop branch, and then pushing another commit.

**Comment by jbathegit** [comment]
> Perhaps. Either way it's saying you need to supply an actual file. What if for example it reads a blank argument. I could beef it up. On another related subject, I noticed that ufbtab makes an error processing a file with msgunp=1, the foreign file case. I made a fix for it. Should I include that with these changes or do a separate fork?

I would suggest a separate issue and PR for the ufbtab fix.

**Comment by jack-woollen** [comment]
@jbathegit I think the wrinkles are sorted out now in the current set of utils updates. It was interesting figuring out how ifort vs gfort works.  Anyway I think the final thing for now is an update of the bufr tar file to get the output of sinv consistent with what comes out of the standard tables in order to pass the sinv test. When that's updated and approval granted, the PR will be complete. 

Also, several other issues came up and occurred to me while working this. 

1) One was the amount of time it takes to download the bufr.tar file from the ftp server. Something like 4 times longer than it takes everything else put together. One solution (short of including it in the repo) is to change it from a tar file to a zip tar file, by using -czf to create it and -xzf to extract it. This shrinks the file by almost 70% and should speedup the d/l quite a bit.

2) For the subroutine getcfmng, a couple of reactions. I like what it does but have 2 suggestions. First would be to blank out the return string, cmeang, similar to filling the output arrays in ufbint. Also how about truncating the output string returned to fit into the output string space provided by the caller. 

3) I could make these changes along with the ufbtab fix if that would work for you.

**Comment by jack-woollen** [comment]
I forgot the mention, after changing sinv.f.in to sinv.f90.in, the configure file macro said it couldn't recognize the compiler. So I just left it as sinv.f.in.

**Comment by jbathegit** [comment]
> @jbathegit I think the wrinkles are sorted out now in the current set of utils updates. It was interesting figuring out how ifort vs gfort works. Anyway I think the final thing for now is an update of the bufr tar file to get the output of sinv consistent with what comes out of the standard tables in order to pass the sinv test. When that's updated and approval granted, the PR will be complete.
> 
> Also, several other issues came up and occurred to me while working this.
> 
> 1. One was the amount of time it takes to download the bufr.tar file from the ftp server. Something like 4 times longer than it takes everything else put together. One solution (short of including it in the repo) is to change it from a tar file to a zip tar file, by using -czf to create it and -xzf to extract it. This shrinks the file by almost 70% and should speedup the d/l quite a bit.
> 2. For the subroutine getcfmng, a couple of reactions. I like what it does but have 2 suggestions. First would be to blank out the return string, cmeang, similar to filling the output arrays in ufbint. Also how about truncating the output string returned to fit into the output string space provided by the caller.
> 3. I could make these changes along with the ufbtab fix if that would work for you.

re: 1, I like this idea so long as it's portable, including to the macOS platform where some of the GitHub CI tests are run.  I think this should be it's own separate issue and PR, but if you want to try that before we try to merge in all of this other stuff, then let me know, including any changes you need to the bufr.tar file, and I can work to get them installed on emcrzdm.  Speaking of which, there were some widespread issues accessing emcrzdm earlier this morning.  I believe they've now been resolved, but I wonder if those might explain the slowness you mentioned you were seeing earlier?  Or have you always noticed this to be an issue?  I haven't generally seen this myself, but either way, and again, I like the idea to do this so long as it's portable, but I do think it should be it's own separate short issue and PR.  Just let me know.

re: 2, and just to clarify, I presume you're suggesting to blank out the return string only in cases where the return value IRET = -1?  If so, then I'm fine with that, but I'm confused by your second suggestion, because I thought the output string was already being automatically truncated by the existing code if the user provided a string space that was less than what was needed to hold the full string.  Or have you found a situation where that wasn't the case?  Only other point I'd add here is that, if possible, any such changes to getcfmng should probably also be their own separate issue and PR, for clarity's sake and to minimize the volume of changes otherwise associated with this particular PR.

re: 3, as noted earlier that's fine, but as a separate issue and PR please ;-)

**Comment by jbathegit** [comment]
> I forgot the mention, after changing sinv.f.in to sinv.f90.in, the configure file macro said it couldn't recognize the compiler. So I just left it as sinv.f.in.

I presume you mean it was complaining when you had a sinv.f90.in file and you tried the following line:
`configure_file(${CMAKE_CURRENT_SOURCE_DIR}/sinv.f90.in ${CMAKE_CURRENT_BINARY_DIR}/sinv.f90 @ONLY)`?  If so, then that's a bit of a stumper, because we already have a similar situation with bfrini.f90.in in the src directory, and that works fine with a similar configure_file command in src/CMakeLists.txt.

**Comment by jack-woollen** [comment]
Re configure file sinv.f90.in: here's what I get
++ cd build
++ cmake -DCMAKE_INSTALL_PREFIX=path1 -DMASTER_TABLE_DIR=/lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork/build/tables ../NCEPLIBS-bufr
-- Configuring done
CMake Error: Cannot determine link language for target "sinv".
CMake Error: CMake can not determine linker language for target: sinv
-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
Jack.Woollen@clogin01:/lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork>

**Comment by jack-woollen** [comment]
Re bufr zip tar file: I put newly made tar and zip tar files in the following directory

Jack.Woollen@clogin01:/lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork/tars> ls -lh 
-rw-r--r-- 1 Jack.Woollen global 241M Feb 25 23:22 bufr-11.6.0.tar
-rw-r--r-- 1 Jack.Woollen global  70M Apr 10 19:18 bufr-11.6.0.tgz
Jack.Woollen@clogin01:/lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork/tars> 

If you put both of these on the rzdm server than my PR can be finished, and then we can also test using the zip tar file as a separate PR.

**Comment by jack-woollen** [comment]
Re the changes to getcfmng. By blanking out the cmeang string I mean ahead of putting anything in it. Just to clear to blanks first then populate it. You may be right about keeping cmeang within bounds in getcfmng. For some reason in my testing I thought there was some problem there. Looking at the source code I don't see a problem, but I also can't tell what is happening in srchtbf either. :-) I'll trust and verify.

**Comment by jack-woollen** [comment]
Re configure file again: this is what the configure file lines looks like:

configure_file(${CMAKE_CURRENT_SOURCE_DIR}/debufr.c.in ${CMAKE_CURRENT_BINARY_DIR}/debufr.c @ONLY)
configure_file(${CMAKE_CURRENT_SOURCE_DIR}/sinv.f90.in ${CMAKE_CURRENT_BINARY_DIR}/sinv.f90 @ONLY)

Maybe something inside sinv.f90.in causing the problem?

**Comment by jack-woollen** [comment]
@jbathegit in case you didn't get emails about the new comments...no hurry, just let you know they're there.

**Comment by jbathegit** [comment]
> Re bufr zip tar file: I put newly made tar and zip tar files in the following directory
> 
> Jack.Woollen@clogin01:/lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork/tars> ls -lh
> -rw-r--r-- 1 Jack.Woollen global 241M Feb 25 23:22 bufr-11.6.0.tar
> -rw-r--r-- 1 Jack.Woollen global 70M Apr 10 19:18 bufr-11.6.0.tgz
> Jack.Woollen@clogin01:/lfs/h2/emc/global/noscrub/Jack.Woollen/jwfork/tars>
> 
> If you put both of these on the rzdm server than my PR can be finished, and then we can also test using the zip tar file as a separate PR.

OK, I just uploaded those files to emcrzdm for you.

**Comment by jbathegit** [comment]
> Re configure file again: this is what the configure file lines looks like:
> 
> configure_file(${CMAKE_CURRENT_SOURCE_DIR}/debufr.c.in ${CMAKE_CURRENT_BINARY_DIR}/debufr.c @ONLY) configure_file(${CMAKE_CURRENT_SOURCE_DIR}/sinv.f90.in ${CMAKE_CURRENT_BINARY_DIR}/sinv.f90 @ONLY)
> 
> Maybe something inside sinv.f90.in causing the problem?

I think the problem here is that you're creating sinv.f90 in the ${CMAKE_CURRENT_BINARY_DIR}, and the compiler can't find it there unless you tell it where it is.  So I think all you need to do is to simply change the line:

`  sinv.f90`

within the CMakeLists.txt _utils_srcs list to instead be:

`  ${CMAKE_CURRENT_BINARY_DIR}/sinv.f90`

and then the configure_file command should work as you have it shown here.

**Comment by jack-woollen** [comment]
@jbathegit it worked to use  ${CMAKE_CURRENT_BINARY_DIR}/sinv.f90. What I don't understand is why it worked without doing that when configuring sinv.f.in. Another cmake cmystery.

File: utils/split_by_subset.f90 (hunk 2 above)
Comment (jbathegit) [reply utils/split_by_subset.f90]: Only remaining comment/concern I have is that the old error message still seems more appropriate for this "if" test.

Looking at some of the other utility codes which have similar logic (e.g. cmpbqm, gettab, readmp), those all do a prior separate `if (file == '')` test with their own bort message similar to the usage diagnostic you now have, followed by a second separate "if" test on the result of the inquire with a bort message similar to the old error message.  So why is this utility any different?

**Comment by jbathegit** [comment]
I just have one remaining comment/concern/question as noted above in the source diffs.  Please let me know if I'm missing something here.

Otherwise, everything else looks good and I'll be happy to approve this PR.

**Comment by jbathegit** [comment]
Thanks for the latest commit @jack-woollen, but I thought we'd agreed that switching over to the zipped tarfile would be its own separate issue and PR?

Also, did you see the note I sent on Friday with the suggestion for the bort message in the split_by_subset.f90 utility?

Finally, please note that a separate PR has now been merged to the develop branch since the last time we spoke.  It doesn't directly affect any of the files in your PR, but you'll still need to rebase your branch with the updated develop branch before we can merge in your changes.

**Comment by jack-woollen** [comment]
Sorry I was just finishing adding the same bort message to each of the utilities at issue. They were all a little different. After looking at the changes needed to implement the zip tar file, I decided you were right when you suggested to include it with the rest of these changes. I thought you'd be okay with it. Anyway I'm about to push the bort message changes now.

**Comment by jack-woollen** [comment]
@jbathegit  I think its good to go now.

**Comment by jbathegit** [comment]
Maybe I'm missing/forgetting something, but I don't remember ever suggesting to include the zip tar changes as part of this PR.  I had always expected it to be its own separate issue and PR.

**Comment by jbathegit** [comment]
That said, it's not a deal-breaker as far as I'm concerned, so I'll go ahead and approve this so you don't have to pull that one piece back out and re-submit it again separately ;-)

**Comment by jack-woollen** [comment]
@jbathegit Quoting from above : re: 1, "I like this idea so long as it's portable, including to the macOS platform where some of the GitHub CI tests are run. I think this should be it's own separate issue and PR, **_but if you want to try that before we try to merge in all of this other stuff, then let me know,_** including any changes you need to the bufr.tar file". I take that as a suggestion, and I was in the process of finishing the update and was about to let you know when you started commenting today. Maybe I misinterpreted your intention. If you prefer I can easily back it out.

**Comment by jack-woollen** [comment]
Great, thanks!

**Comment by jbathegit** [comment]
> @jbathegit Quoting from above : re: 1, "I like this idea so long as it's portable, including to the macOS platform where some of the GitHub CI tests are run. I think this should be it's own separate issue and PR, **_but if you want to try that before we try to merge in all of this other stuff, then let me know,_** including any changes you need to the bufr.tar file". I take that as a suggestion, and I was in the process of finishing the update and was about to let you know when you started commenting today. Maybe I misinterpreted your intention. If you prefer I can easily back it out.

You did misunderstand me.  When I said, if you want to try "that", what I meant was if you want to submit a separate PR to merge the zip tar change into the develop branch, before submitting this PR to merge all of the utils tests and fixes into the develop branch, then let me know.

Again, no worries and not a big deal.  I'll go ahead and approve the whole package now.

**Review by jbathegit — APPROVED** [review]

**Pull request merged**